sys/linux/socket.txt

```diff
@@ -317,6 +321,41 @@ ifr_ifru [
 
 define IFREQ_SIZE	sizeof(struct ifreq)
 
+# https://elixir.bootlin.com/linux/v5.3.11/source/include/uapi/linux/if_vlan.h
+vlan_args {
+	cmd		flags[vlan_cmd, intptr]
+	device1		array[int8, 24]
+	u		vlan_u
+	vlan_qos	int8
+}
+
+vlan_u [
+	device2		array[int8, 24]
+	VID		intptr
+	skb_priority	intptr
+	name_type	intptr
+	bind_type	intptr
+	flag		intptr
+]
+
+vlan_cmd = ADD_VLAN_CMD, DEL_VLAN_CMD, SET_VLAN_INGRESS_PRIORITY_CMD, SET_VLAN_EGRESS_PRIORITY_CMD, GET_VLAN_INGRESS_PRIORITY_CMD, GET_VLAN_EGRESS_PRIORITY_CMD, SET_VLAN_NAME_TYPE_CMD, SET_VLAN_FLAG_CMD, GET_VLAN_REALDEV_NAME_CMD, GET_VLAN_VID_CMD
+
+# https://elixir.bootlin.com/linux/v5.3.12/source/include/uapi/linux/route.h#L31
+rtentry {
+	rt_pad1		intptr
+	rt_dst		sockaddr
+	rt_gateway	sockaddr
+	rt_genmask	sockaddr
+	rt_flags	int16
+	rt_pad2		int16
+	rt_pad3		intptr
+	rt_pad4		ptr[in, intptr]
+	rt_metric	int16
+	rt_dev		ptr[in, int8]
```

Comment:
This is actually `ptr[in, devname, opt]`. The raw C types do not necessary capture all relevant semantic info, we need that semantic info, not just raw C types. So purely mechanical transformation is frequently not enough (if would need just mechanical transformation, we would auto extract these descriptions from header files, but unfortunately the headers are simply incomplete).

Reply:
This is not purely mechanical transformation, I did a artificial transformation here and sorry I wasn't aware there is a devname at the moment, and I should have noticed that it is optional, will take care of it next time.